docs/blog/2019-11-19-gatsby-brings-speed-to-mediacurrent-com/index.md

```diff
@@ -0,0 +1,70 @@
+  - To resolve this, we started using a Jenkins build process which does the build, then pushes the build to Netlify. Since caching is still available on Jenkins, the build times have improved.
+  - More recently, we are working on leveraging [Gatsby Preview](/preview/) which allows us to see changes _immediately_ on a preview site as well as the Gatsby Builds feature which cuts down our build times dramatically.
+- Image processing
+  - Processing images was one of the build processes that increased the build time. Consequently, many published images caused the build to be longer than expected. We worked on reducing and unpublishing older content and redirecting it to new, fresh content.
+- Body field processing.
+  - Images in the body field are not part of the regular processing and Drupal will render relative paths. Since these paths are not part of Gatsby, we needed to add logic that updates an image’s path to point to our Drupal instance URL.
+  - While we can access the images through the entity id, there was a second problem where the file would not transliterate the names and add the “%20” HTML character code where spaces are expected. At this time we are working on a patch for the gatsby-source-drupal, which changes the name of the file to replace %20 with a dash to make it more browser friendly. This will be contributed back to the community. In the meantime, we are redirecting any images inside a body field to our publishing system.
+- We found short codes for files from our Drupal 7 instance that were processed in the Jekyll build process.
+  - We band-aided this fix with our Jekyll module. We had to go back and re-publish this information correctly.
+- Audio file information.
+  - This was patched through our Jekyll module, but we added a JSON:API field enhancer which added the necessary data to show audio files.
+- External JS calls. (Acquia Lift, Pardot)
+  - Since Gatsby is ultimately a React application, the way to insert tracking cookies, and other 3rd Party scripts required workarounds. Fortunately, the gatsby-ssr API made this easy. We used the onRenderBody lifecycle to confirm which environment we were on, and inserted the required script tags accordingly.
+
+## Final Thoughts
+
+Ultimately, the decision to convert Mediacurrent.com to a Gatsby front-end was a wise move. Considering the issues we inherited by using a static-site generator that wasn’t intentionally built as a CMS front-end, Gatsby has been a welcome improvement. The benefits we’re experiencing with Gatsby are:
+
+- No wait time to apply Drupal module updates. The Gatsby front-end and Drupal back-end are completely decoupled.
+- Simplified decoupled architecture with no middle-men.
+- Front-end design updates can be deployed without concern for impact on back-end logic.
+- [Gatsby Preview](/preview/) allows us to view content updates in real-time prior to publishing to production.
```

Comment:
this should link to https://www.gatsbyjs.com/preview/

Reply:
Updated